pumps/resurface.go

```diff
@@ -0,0 +1,234 @@
+package pumps
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"github.com/resurfaceio/logger-go/v3"
+)
+
+type ResurfacePump struct {
+	logger *logger.HttpLogger
+	config *ResurfacePumpConfig
+	CommonPumpConfig
+}
+
+type ResurfacePumpConfig struct {
+	EnvPrefix string `mapstructure:"meta_env_prefix"`
+	Url       string `mapstructure:"capture_url"`
+	Rules     string
+	Queue     []string
+}
+
+const (
+	resurfacePrefix     = "resurface-pump"
+	resurfacePumpName   = "Resurface Pump"
+	resurfaceDefaultEnv = PUMPS_ENV_PREFIX + "_RESURFACEIO" + PUMPS_ENV_META_PREFIX
+)
+
+func (rp *ResurfacePump) New() Pump {
+	newPump := ResurfacePump{}
+	return &newPump
+}
+
+func (rp *ResurfacePump) GetName() string {
+	return resurfacePumpName
+}
+
+func (rp *ResurfacePump) GetEnvPrefix() string {
+	return rp.config.EnvPrefix
+}
+
+func (rp *ResurfacePump) Init(config interface{}) error {
+	rp.config = &ResurfacePumpConfig{}
+	rp.log = log.WithField("prefix", resurfacePrefix)
+
+	err := mapstructure.Decode(config, &rp.config)
+	if err != nil {
+		rp.log.Debug("Failed to decode configuration: ", err)
+		return err
+	}
+
+	processPumpEnvVars(rp, rp.log, rp.config, resurfaceDefaultEnv)
+
+	opt := logger.Options{
+		Rules: rp.config.Rules,
+		Url:   rp.config.Url,
+		Queue: rp.config.Queue,
+	}
+	rp.logger, err = logger.NewHttpLogger(opt)
+	if err != nil {
+		rp.log.Error(err)
+		return err
+	}
+	if rp.logger.Enabled() {
+		rp.log.Info(rp.GetName() + " Initialized")
+		return nil
+	} else {
+		rp.log.Info(rp.GetName() + " Initialized (Logger disabled)")
+		return errors.New("logger is not enabled")
+	}
+}
+
+func parseHeaders(headersString string, existingHeaders http.Header) (headers http.Header) {
+	if existingHeaders != nil {
+		headers = http.Header.Clone(existingHeaders)
+	} else {
+		headers = http.Header{}
+	}
+	for _, line := range strings.Split(headersString, "\r\n") {
+		header := strings.Split(line, ": ")
+		if len(header) < 2 {
+			continue
+		}
+		headers.Add(header[0], header[1])
+	}
+	return
+}
+
+func mapRawData(rec analytics.AnalyticsRecord) (httpReq http.Request, httpResp http.Response, customFields map[string]string, err error) {
```

Comment:
@monrax we should pass the `analytics.AnalyticsRecord` as a pointer here, it's a heavy struct.